pennylane/ops/functions/matrix.py

```diff
@@ -134,13 +134,27 @@ def _matrix(tape, wire_order=None):
     interface = qml.math._multi_dispatch(params)
 
     wire_order = wire_order or tape.wires
-    n_wires = len(wire_order)
 
     # initialize the unitary matrix
-    unitary_matrix = qml.math.eye(2**n_wires, like=interface)
+    unitary_matrix = qml.math.eye(2 ** len(wire_order), like=interface)
 
+    broadcasted_unitary = False

```

Comment:
Minor: suggest renaming to differentiate bwn. unitary of next op and the unitary whose matrix this method returns.
```suggestion
    broadcasted_result = False
```

Reply:
Should we rename `unitary_matrix` then as well? :)

Reply:
I went for `result` instead of `unitary_matrix` and for `result_is_broadcasted`, in analogy to the below `U_is_broadcasted`.

Reply:
Good question. That renaming is out of scope here. The clean way would be to rename that in a separate PR. Having said that, if we do want to make the change, I think we could do that here (provided it's a minor scope change, which it seems to be).

Reply:
Sorry, my comments were batched up in a review, hence the weird order. I think the scope is very small here, and it doesn't make sense to open a PR for this, imho.